### PR TITLE
[Improvement]: Add an AMS configuration property to skip table creation when AMS startup

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

Our RDS doesn't have a information_schema 💔

## Brief change log

- Add an AMS property `ams.database.init-schema`, default value is true

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes )
- If yes, how is the feature documented? (docs )

### CODE REVIEW

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroManagementConf.java [hunk 1]
```diff
@@ -260,6 +260,12 @@ public class AmoroManagementConf {
           .defaultValue("jdbc:derby:/tmp/amoro/derby;create=true")
           .withDescription("Database connection address");
 
+  public static final ConfigOption<Boolean> DB_INIT_SCHEMA =
+      ConfigOptions.key("database.init-schema")
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroManagementConf.java:264]: Is `database.auto-create-tables` a proper name?

Comment (zhoujinsong) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroManagementConf.java:264]: BTW, we may add this new configuration to our example configuration file.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/SqlSessionFactoryProvider.java [hunk 2]
```diff
@@ -123,6 +126,11 @@ public void init(Configurations config) {
    * @param config
    */
   private void createTablesIfNeed(Configurations config) {
+    boolean initSchema = config.getBoolean(AmoroManagementConf.DB_INIT_SCHEMA);
+    if (!initSchema) {
+      LOG.info("Skip database schema initialize.");
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/SqlSessionFactoryProvider.java:131]: I recommend adding more log outputs for each situation here: 

- the table was created automatically
- the table already exists so there is no need to create it automatically
- automatic creation is not enabled, so the table was not created automatically.

File: docs/admin-guides/deployment.md [hunk 3]
```diff
@@ -129,10 +129,13 @@ ams:
 #    type: postgres
 #    jdbc-driver-class: org.postgresql.Driver
 #    url: jdbc:postgresql://127.0.0.1:5432/amoro
+#    init-schema: false
 #    username: user
 #    password: passwd
 ```
 
+AMS will create tables automatically, if you want to create tables on yourself, set `ams.database.init-schema` to false.
```
Comment (zhoujinsong) [inline docs/admin-guides/deployment.md:137]: You can combine the description with line 113.

**Review by zhoujinsong — COMMENTED** [review]
I left some comments, PTAL.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroManagementConf.java (hunk 1 above)
Comment (baiyangtx) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroManagementConf.java]: fixed

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/SqlSessionFactoryProvider.java (hunk 2 above)
Comment (baiyangtx) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/SqlSessionFactoryProvider.java]: fixed

File: docs/admin-guides/deployment.md (hunk 3 above)
Comment (baiyangtx) [reply docs/admin-guides/deployment.md]: fixed

**Review by zhoujinsong — APPROVED** [review]
LGTM.
